code-challenges.test.js

```diff
@@ -14,12 +16,45 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("codedMessage", () => {
+    it("takes in a string and returns a string with a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0", () => {
+        const secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c" 
+      expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+      
+    })
+  }) 
+
+  const codedMessage = (str) => {                    //create a function called codedMessage that takes in a string 
+    let newStr = ""                                  // define a new empty string 
+    for(let i = 0; i < str.length; i++) {            // create a for loop checking length of string 
+        if(str[i] === "a" || str[i] === "A") {          //if string is equal to "a" or "A" add the number 4 to newStr
+            newStr += "4"
+        }
+        else if(str[i] === "e" || str[i] === "E") {     //else if str is equal to "e" or "E" add the number 3 to newStr
+            newStr += "3"
+        }
+        else if(str[i] === "i" || str[i] === "I") {     //else if str is equal to "i" or "I" add the number 1 to newStr
+            newStr += "1"
+        }
+        else if(str[i] === "o" || str[i] === "O") {     //else if str is equal to "o" or "O" add the number 0 to newStr
+            newStr += "0"
+        }
+        else {
+            newStr += str[i]                               //else store current value to the newStr
+        }
+    }
+    return newStr
```

Comment:
I like the line for line pseudo code. As a further challenge, how can you refactor this without needing to create the newStr variable? Most of the time, these kinds of variables are not needed. There are methods available to use that work on strings and also return a string, so how can you put the rest of the logic in one of those methods to get this same result?